docs/src/man/querying_frameworks.md

```diff
@@ -0,0 +1,129 @@
+
+To get started, install the Query.jl package:
+
+```julia
+Pkg.add("Query")
+```
+
+A query is started with the `@from` macro and consists of a series of query commands. Query.jl provides commands that can filter, project, join, group, flatten and group data from a `DataFrame`. A query can return an iterator, or one can materialize the results of a query into a variety of data structures, including a new `DataFrame`.
+
+A simple example of a query looks like this:
+
+```@meta
+DocTestSetup = quote
+    using DataFrames, Query
+end
+```
+
+```jldoctest
+using DataFrames, Query
+
+df = DataFrame(name=["John", "Sally", "Roger"], age=[54., 34., 79.], children=[0, 2, 4])
+
+q1 = @from i in df begin
+     @where i.age > 40
+     @select {number_of_children=i.children, i.name}
+     @collect DataFrame
+end
+
+println(q1)
+
+# output
+
+2×2 DataFrames.DataFrame
+│ Row │ number_of_children │ name    │
+├─────┼────────────────────┼─────────┤
+│ 1   │ 0                  │ John  │
+│ 2   │ 4                  │ Roger │
+```
+
+```@meta
+DocTestSetup = nothing
+```
+
+The query starts with the `@from` macro. The first argument `i` is the name of the range variable that will be used to refer to an individual row in later query commands. The next argument `df` is the data source that one wants to query. The `@where` command in this query will filter the source data by applying the filter condition `i.age > 40`. This filters out any rows in which the `age` column is not larger than 40. The `@select` command then projects the columns of the source data onto a new column structure. The example here applies three specific modifications: 1) it only keeps a subset of the columns in the source `DataFrame`, i.e. the `age` column will not be part of the transformed data; 2) it changes the order of the two columns that are selected; and 3) it renames one of the columns that is selected from `children` to `number_of_children`. The example query uses the `{}` syntax to achieve this. A `{}` in a Query.jl expression instantiates a new [NamedTuple](https://github.com/blackrock/NamedTuples.jl), i.e. it is a shortcut for writing `@NT(number_of_children=>i.children, name=>i.name)`. The `@collect` statement determines the data structure that the query returns. In this example the results are returned as a `DataFrame`.
+
+A query without a `@collect` statement returns a standard julia iterator that can be used with any normal julia language construct that can deal with iterators. The following code returns a julia iterator for the query results:
+
+```julia
+q2 = @from i in df begin
+     @where i.age > 40
+     @select {number_of_children=i.children, i.name}
+end
+```
+
+One can loop over the results using a standard julia `for` statement:
+
+```@meta
+DocTestSetup = quote
+    using DataFrames, Query
```

Comment:
I don't think so. Each jldoctest block is run independently and no state is preserved between these code blocks. I guess the only alternative would be to repeat the query itself in the jldoctest, and only have the ``using`` and DataFrame construction code in the ``@meta`` section.